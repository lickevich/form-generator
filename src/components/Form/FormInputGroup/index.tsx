import React from "react";

import { IInput } from "../../../type";
import Row from "../Row";
import Col from "../Col";

const FormInputGroup: React.FC<IInput> = ({
  type = "",
  groupLabel = "",
  items = [],
}) => {
  return (
    <Row>
      <Col>
        <label title={groupLabel}>{groupLabel}</label>
      </Col>
      <Col>
        {items.map(({ label = "", ...rest }, index) => {
          const id = `${label}${index}`;

          return (
            <div key={id}>
              <input id={id} type={type} {...rest} />
              <label htmlFor={id} title={label}>
                {label}
              </label>
            </div>
          );
        })}
      </Col>
    </Row>
  );
};

export default FormInputGroup;
