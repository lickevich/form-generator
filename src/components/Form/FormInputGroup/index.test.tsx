import React from "react";
import { render } from "@testing-library/react";

import FormInputGroup from "./";

describe("FormInputGroup", () => {
  const props = {
    type: "radio",
    groupLabel: "radio",
    items: [
      { label: "radio1", name: "radio", value: "1" },
      { label: "radio2", name: "radio", value: "2" },
    ],
  };

  it("render the component", () => {
    const component = <FormInputGroup {...props} />;
    const { asFragment } = render(component);

    // @ts-ignore
    expect(asFragment(component)).toMatchSnapshot();
  });
});
