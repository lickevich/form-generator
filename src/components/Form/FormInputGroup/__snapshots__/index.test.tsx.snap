// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormInputGroup render the component 1`] = `
<DocumentFragment>
  <div
    class="sc-bdvvaa kTYFsi"
  >
    <div
      class="sc-gsDJrp fhKCEr"
    >
      <label
        title="radio"
      >
        radio
      </label>
    </div>
    <div
      class="sc-gsDJrp fhKCEr"
    >
      <div>
        <input
          id="radio10"
          name="radio"
          type="radio"
          value="1"
        />
        <label
          for="radio10"
          title="radio1"
        >
          radio1
        </label>
      </div>
      <div>
        <input
          id="radio21"
          name="radio"
          type="radio"
          value="2"
        />
        <label
          for="radio21"
          title="radio2"
        >
          radio2
        </label>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
